src/main/java/com/ericsson/ei/mongodbhandler/MongoDBHandler.java

```diff
@@ -298,7 +298,16 @@ public void createTTLIndex(String dataBaseName, String collectionName, String fi
         if (!collectionList.contains(collectionName)) {
             log.debug("The requested database(" + dataBaseName + ") / collection(" + collectionName
                     + ") not available in mongodb, Creating ........");
-            db.createCollection(collectionName);
+            try {
+                db.createCollection(collectionName);                
+            } catch (MongoCommandException e) {
+                String message = "collection '" + dataBaseName + "." + collectionName + "' already exists";
```

Comment:
I guess the Exception can be caused by other errors than " ..... already exist." . So a more suitable message should be "Failed to create collection: " + collectionName  + " \nError message: " + e.getMessage()....

Reply:
The warning message is only sent if the error message contains ("collection 'dataBaseName.collectionName' already exists") All other errors caught here will be sent further.

Reply:
Ok,, if the error message for user is changed, log.warn will then never be executed/logged.